src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -147,9 +146,7 @@ class AuthScreens extends React.Component {
             nvpNames: ONYXKEYS.NVP_PREFERRED_LOCALE,
         }).then((response) => {
             const preferredLocale = lodashGet(response, ['nameValuePairs', 'preferredLocale'], CONST.DEFAULT_LOCALE);
-            if ((currentPreferredLocale !== CONST.DEFAULT_LOCALE) && (preferredLocale !== currentPreferredLocale)) {
-                setLocale(currentPreferredLocale);
-            } else {
+            if (preferredLocale !== currentPreferredLocale) {
```

Comment:
I think the issue is because the user is the user is not in beta and cannot see language settings inside the app.
This condition was used to cover the case - where user has set up preferred language as `es`,and they open app on new device/incognito window, the login page would appear in english and if they're familiar with language, they'll login to the app. At this time we don't change their preferred locale NVP.

Reply:
Just FYI, This is also being refactored [here](https://github.com/Expensify/App/pull/5943)

Reply:
I tested on dev where all users are in the beta. I think the main thing this is getting rid of is that `setLocale` also changes the user's preferred locale NVP, which is something we should not explicitly set without a user requesting a language change.

For example: If I login as an `en` user, then login to an `es` user then switch back to the `en` user then the `en` user would have their settings in `es` without ever having touched their language settings. I understand the refactoring would help with the login page, but I still feel we should be getting rid of `setLocale` here regardless.

Reply:
>  If I login as an `en` user, then login to an `es` user then switch back to the `en` user then the `en` user would have their settings in `es` without ever having touched their language settings.

Yea, this is a problem for sure. It looks like that other PR is tackling the same issue though. Are we not doubling up on this?

Reply:
This PR is meant to simply squash the bug and looking at the other PR I'm not sure where that one is headed in terms of refactoring. I'm sure there will be more work to complete the language feature settings, but considering this is still a real bug that can affect users I'm in favor of merging for now (https://github.com/Expensify/App/pull/5943#issuecomment-958582692)